### PR TITLE
[Hackney] Add cobrand & assets groundwork

### PR DESCRIPTION
**Pull request by davea**
[skip changelog]

### CODE REVIEW

File: bin/hackney/.gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+venv
```
Comment (dracos) [inline bin/hackney/.gitignore:1]: Would you mind having the python bit of this in the fixmystreet.com repo? Just as it's totally separate, and that's deployed alongside anyway.

File: bin/hackney/import_categories [hunk 2]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env perl
```
Comment (dracos) [inline bin/hackney/import_categories:1]: I see a couple of small differences, but is it worth factoring this together with the main bin/import_categories script, perhaps by adding a couple of hooks and the like? For the next time?

File: bin/hackney/parse_categories_sheet.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
```
Comment (dracos) [inline bin/hackney/parse_categories_sheet.py:1]: Afraid I haven't run this, but I assume it produces the output JSON you want :)

File: web/cobrands/fixmystreet-uk-councils/alloy.js [hunk 4]
```diff
@@ -133,4 +139,59 @@ fixmystreet.alloy_defaults = {
...
+    // the layer uses EPSG:3857 for generating the tile location but the features
+    // use EPSG:4326
+    getRemoteProjection: function() {
+        return this.remote;
+    }
+});
+
+fixmystreet.alloyv2_defaults = {
+    http_options: {
+      base: "https://tilma.staging.mysociety.org/resource-proxy/proxy.php?https://northants.assets/${layerid}/${x}/${y}/${z}/cluster?styleIds=${styleid}"
```
Comment (dracos) [inline web/cobrands/fixmystreet-uk-councils/alloy.js:191]: This URL doesn't feel like it should be in a defaults object :)

File: web/cobrands/fixmystreet-uk-councils/alloy.js [hunk 5]
```diff
@@ -133,4 +139,59 @@ fixmystreet.alloy_defaults = {
     strategy_class: OpenLayers.Strategy.Alloy
 };
 
+/* for Alloy V2 */
+OpenLayers.Format.Alloy = OpenLayers.Class(OpenLayers.Format.GeoJSON, {
```
Comment (dracos) [inline web/cobrands/fixmystreet-uk-councils/alloy.js:143]: Do we want to call it AlloyV2 for consistency with other classes?

File: web/cobrands/hackney/assets.js [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+      }
+  },
+  asset_category: "Overflowing bin ",
+  attributes: {}
+});
+
+fixmystreet.assets.add(wfs_defaults, {
+  http_options: {
+      params: {
+          TYPENAME: "amenity:public_toilet",
```
Comment (dracos) [inline web/cobrands/hackney/assets.js:43]: Toilets appear again further down? Bit confused as to the categories here.

File: web/cobrands/hackney/assets.js [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+};
+
+
+fixmystreet.assets.add(wfs_defaults, {
+  http_options: {
+      params: {
+          TYPENAME: "greenspaces:park_bin",
+      }
+  },
+  asset_category: "Overflowing bin ",
```
Comment (dracos) [inline web/cobrands/hackney/assets.js:36]: Final space needed?

File: web/cobrands/hackney/assets.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+  attributes: {}
+});
+
+fixmystreet.assets.add(wfs_defaults, {
+  http_options: {
+      params: {
+          TYPENAME: "recycling:estate_recycling_bin",
+      }
+  },
+  asset_category: "Bin Contamination (2)",
```
Comment (dracos) [inline web/cobrands/hackney/assets.js:66]: Another (2)

File: web/cobrands/hackney/assets.js [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+    }
+  }
+});
+
+$.each(layers, function(index, layer) {
+    if ( layer.categories && layer.styleid ) {
+        var options = {
+          http_options: {
+            base: "https://tilma.staging.mysociety.org/resource-proxy/proxy.php?https://hackney.assets/${layerid}/${x}/${y}/${z}/cluster?styleIds=${styleid}",
+            styleid: layer.styleid,
```
Comment (dracos) [inline web/cobrands/hackney/assets.js:180]: what does the styleid do? Also why is it both here and in the base as styleIds?

File: web/cobrands/hackney/assets.js [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+            layerid: layer.layerid,
+          },
+          asset_type: layer.asset_type || "spot",
+          asset_category: layer.categories,
+          asset_item: layer.item_name || layer.layer_name.toLowerCase(),
```
Comment (dracos) [inline web/cobrands/hackney/assets.js:185]: Final comma

File: web/cobrands/hackney/assets.js [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+  non_interactive: false,
+  body: "Hackney Council",
+  attributes: {
+    asset_resource_id: function() {
+      return this.fid;
+    }
+  }
+});
+
+$.each(layers, function(index, layer) {
```
Comment (dracos) [inline web/cobrands/hackney/assets.js:175]: Worth factoring this in with the very similar code in northamptonshire?

File: templates/web/hackney/header_extra.html [hunk 12]
```diff
@@ -0,0 +1,2 @@
+[% INCLUDE 'tracking_code.html' %]
```
Comment (dracos) [inline templates/web/hackney/header_extra.html:1]: This file is missing

File: templates/web/hackney/about/privacy.html [hunk 13]
```diff
@@ -0,0 +1,388 @@
+[% INCLUDE 'header.html',
```
Comment (dracos) [inline templates/web/hackney/about/privacy.html:1]: If the only change from fixmystreet-uk-councils is the basis for processing, then I suggest we factor the templates together somehow.

File: t/cobrand/hackney.t [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+    description => 'question',
+    variable => 'true',
+    required => 'false',
+    order => 1,
+    datatype_description => 'datatype',
+} ) );
+$contact->update;
+
+my $user = $mech->create_user_ok('user@example.org', name => 'Test User');
+my $hackney_user = $mech->create_user_ok('iow_user@example.org', name => 'Hackney User', from_body => $hackney);
```
Comment (dracos) [inline t/cobrand/hackney.t:46]: strange email ;)

File: perllib/FixMyStreet/Cobrand/Hackney.pm [hunk 15]
```diff
@@ -55,4 +55,12 @@ sub default_map_zoom { 5 }
 
 sub admin_user_domain { 'hackney.gov.uk' }
 
+sub anonymous_account {
```
[the remark this answers is not in the document]
Comment (davea) [reply perllib/FixMyStreet/Cobrand/Hackney.pm:58]: It'll use `Default::allow_anonymous_reports` for the per-category anonymous reporting.

File: bin/hackney/.gitignore (hunk 1 above)
Comment (davea) [reply bin/hackney/.gitignore:1]: That's a good idea. It's a one-time thing anyway and has already served its purpose (as we'll be promoting the categories from staging to live).

File: perllib/FixMyStreet/Cobrand/Hackney.pm (hunk 15 above)
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Hackney.pm:58]: Forgot I'd added that!

**Pull request merged**